src/gcp/cloudfunctions.ts

```diff
@@ -260,6 +251,74 @@ export async function setIamPolicy(options: IamOptions): Promise<void> {
   }
 }
 
+/**
+ * Formats the service account to be used with IAM API calls
+ * @param serviceAccount the custom service account created by the user
+ * @param projectId the ID of the current project
+ * @returns a correctly formatted service account string
+ */
+function formatServiceAccount(serviceAccount: string, projectId: string): string {
+  if (serviceAccount.length === 0) {
+    throw new Error("Service account cannot be an empty string");
+  }
+  let emailId = "";
+  if (serviceAccount.charAt(serviceAccount.length - 1) === "@") {
```

Comment:
if you want to be fancy, you can use a ternary here:
```
const emailId = (serviceAccount.charAt(serviceAccount.length - 1) === "@")
  ? `${serviceAccount}${projectId}.iam.gserviceaccount.com` 
  : `${serviceAccount}@${projectId}.iam.gserviceaccount.com`;
```

Reply:
Switched to the ternary operator